<a href="https://colab.research.google.com/github/alysonbz/lab_prog_si/blob/gabriel_coelho/atividade2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

1) Complete o código. Observe que a variável bath (float) é precedida por uma string “bathroom”. O mesmo acontece com liv. Complete atribuindo as strings "hallway", "kitchen" e "bedroom" na lista nos locais apropriados.

1.1 ) Print a areas na linha 11.

In [ ]:
#AREA VARIABLES (IN SQUARE METERS)
hall=11.25
kit=18.0
liv=20.0
bed=10.75
bath=9.5

#ADAPT LIST AREAS
areas=["hallway",hall,"kitchen",kit,"living room",liv,"bedroom",bed,"bathroom",bath]

#PRINT AREAS
print(areas)

['hallway', 11.25, 'kitchen', 18.0, 'living room', 20.0, 'bedroom', 10.75, 'bathroom', 9.5]


2) Suponha que as seguintes listas foram criadas.

A. [1, 3, 4, 2] B. [[1, 2, 3], [4, 5, 7]] C. [1 + 2, 'a' * 5, 3]

Qual forma de criação de lista está correta? Descreva o que contém nas listas definidas corretamente.

*RESP:*

A. - lista está correta, pois apresenta números inteiros.

B. - lista está correta, pois representa uma lista com outra lista dentro e cada uma delas apresenta números inteiros.

C. - lista está correta, pois contém uma expressão. seu resultado final, se executada, será: [3, "aaaaa", 3].

3) Complete a lista de listas com as informações de  bedroom e bathroom.

3.1) Mostre a variável house.

3.2) Print o tipo da variável house.


In [1]:
#AREA VARIABLES (IN SQUARE METERS)
hall=11.25
kit=18.0
liv=20.0
bed=10.75
bath=9.5

#HOUSE INFORMATION AS LIST OF LISTS
house=[["hallway",hall],
       ["kitchen",kit],
       ["living room",liv],
       ["bedroom",bed],
       ["bathroom",bath]]

#PRINT HOUSE
print(house)

#PRINT TYPE HOUSE
print(type(house))

[['hallway', 11.25], ['kitchen', 18.0], ['living room', 20.0], ['bedroom', 10.75], ['bathroom', 9.5]]
<class 'list'>


4) Exiba o segundo elemento da lista

4.1) Exiba o último elemento da lista usando index negativo.

4.2) Exiba o elemento que indica o valor da área de Livin room.

In [3]:
#CREATE THE AREAS LIST
areas=["hallway",11.25,"kitchen",18.0,"living room",20.0,"bedroom",10.75,"bathroom",9.5]

#PRINT OUT SECOND ELEMENT FROM AREAS
print(areas[1])

#PRINT OUT LAST ELEMENT FROM AREAS
print(areas[5])

#PRINT OUT THE AREA OF THE LIVING ROOM
print(areas[3])

11.25
20.0
18.0


5) Na linha 7, acessando os elementos da lista, realize a soma dos valores de Kitchen e living room e atribua em eat_sleep_area.

5.1) Exiba a variável eat_sleep_area.

In [4]:
#CREATE THE AREAS LIST
areas=["hallway",11.25,"kitchen",18.0,"living room",20.0,"bedroom",10.75,"bathroom",9.5]

#SUM OF KITCHEN AND BEDROOM AREA: EAT_SLEEP_AREA
eat_sleep_area=areas[3]+areas[7]

#PRINT THE VARIABLE EAT_SLEEP_AREA
print(eat_sleep_area)

28.75


6) Acesse os 6 primeiros elemento de areas e atribua a uma nova lista: downstairs.

6.1) Acesse os 4 últimos elemento de areas e atribua a uma nova lista: upstairs.

6.2) Exiba downstairs e upstairs.

In [5]:
#CREATE THE AREAS LIST
areas=["hallway",11.25,"kitchen",18.0,"living room",20.0,"bedroom",10.75,"bathroom",9.5]

#USE SLICING TO CREAT DOWNSTAIRS
downstairs=areas[0:6]

#USE SLICING TO CREAT UPSTAIRS
upstairs=areas[6:10]

#PRINT DOWNSTAIRS AND UPSTAIRS
print(downstairs)
print(upstairs)

['hallway', 11.25, 'kitchen', 18.0, 'living room', 20.0]
['bedroom', 10.75, 'bathroom', 9.5]


7) Troque o valor de bathroom de 9.50 para 10.50.

7.1) Substitua a string “Living room” por “chill zone”

In [24]:
#CREATE THE AREAS LIST
areas=["hallway",11.25,"kitchen",18.0,"living room",20.0,"bedroom",10.75,"bathroom",9.5]

#CORRECT THE BATHROOM AREA
areas[9]=10.50
areas

#CHANGE "LIVING ROOM" TO "CHILL ZONE"
areas[4]='chill zone'
areas

['hallway',
 11.25,
 'kitchen',
 18.0,
 'chill zone',
 20.0,
 'bedroom',
 10.75,
 'bathroom',
 10.5]

8) Adicione ["poolhouse", 24.5] no fim da lista  e atribua em uma nova lista chamada de areas_1

8.1) 8) Adicione ["garage", 15.5] no fim da lista areas_1  e atribua em uma nova lista chamada de areas_2

8.2) Exiba areas_2

In [33]:
#CREATE THE AREAS LIST
areas=['hallway',11.25,'kitchen',18.0,'chill zone',20.0,'bedroom',10.75,'bathroom',10.5]

#ADD POOLHOUSE DATA TO AREAS, NEW LIST IS AREAS_1
areas_1=areas+['poolhouse',24.5]
areas_1

#ADD GARAGE DATA TO AREAS_1, NEW LIST IS AREA_2
areas_2=areas_1+['garage',15.5]
areas_2

['hallway',
 11.25,
 'kitchen',
 18.0,
 'chill zone',
 20.0,
 'bedroom',
 10.75,
 'bathroom',
 10.5,
 'poolhouse',
 24.5,
 'garage',
 15.5]

9) Execute o código e observe que a modificação realizada em areas_copy afetou areas. Corrija esse código e exiba areas e areas_copy no fim do código.

In [37]:
#CREATE THE AREAS LIST
areas=[11.25, 18.0, 20.0, 10.75, 9.5]

#CREATE AREAS_COPY
areas_copy=areas

#CHANGE AREAS_COPY
areas_copy=areas+[5.0]

#PRINT AREAS
print(areas)
print(areas_copy)

[11.25, 18.0, 20.0, 10.75, 9.5]
[11.25, 18.0, 20.0, 10.75, 9.5, 5.0]
